### PR TITLE
More test speedups via subtest + removal rather than skipping

### PR DESCRIPTION
**Pull request by reaperhulk**
Split into #5618, #5619, #5620 

This will be rebased to be just the EC tests (which appear to have coverage issues atm) after those three land.

### CODE REVIEW

File: tests/hazmat/primitives/utils.py [hunk 1]
```diff
@@ -76,6 +76,8 @@ def generate_aead_test(
     param_loader, path, file_names, cipher_factory, mode_factory
 ):
     all_params = _load_all_params(path, file_names, param_loader)
+    # We don't support IVs < 64-bit so just strip them out
+    all_params = [i for i in all_params if len(i["iv"]) >= 16]
```
Comment (alex) [inline tests/hazmat/primitives/utils.py:80]: Is this true for all AEADs?

Comment (reaperhulk) [reply tests/hazmat/primitives/utils.py:80]: No, should be constrained to GCM. Pushing fix

**Review by alex — APPROVED** [review]

**Comment by alex** [comment]
rebase now

File: tests/hazmat/primitives/test_ec.py [hunk 2]
```diff
@@ -505,40 +498,39 @@ def test_load_invalid_public_ec_key_from_numbers(self, backend):
...
-            vector["x"], vector["y"], curve_type()
-        ).public_key(backend)
+                key = ec.EllipticCurvePublicNumbers(
+                    vector["x"], vector["y"], curve_type()
+                ).public_key(backend)
 
-        signature = encode_dss_signature(vector["r"], vector["s"])
+                signature = encode_dss_signature(vector["r"], vector["s"])
 
         if vector["fail"] is True:
```
Comment (alex) [inline tests/hazmat/primitives/test_ec.py:535]: This needs to be indented

**Review by alex — APPROVED** [review]

**Pull request merged**